ext/compressors/snappy/snappy_compress.c

```diff
@@ -108,91 +108,78 @@ wt_snappy_compress(WT_COMPRESSOR *compressor, WT_SESSION *session,
 	char *snapbuf;
 
 	/*
-	 * dst_len was computed in wt_snappy_pre_size, so we know it's big
-	 * enough.  Skip past the space we'll use to store the final count
-	 * of compressed bytes.
+	 * dst_len was computed in snappy_pre_size, so we know it's big enough.
+	 * Skip past the space we'll use to store the final count of compressed
+	 * bytes.
 	 */
 	snaplen = dst_len - sizeof(size_t);
 	snapbuf = (char *)dst + sizeof(size_t);
 
 	/* snaplen is an input and an output arg. */
 	snret = snappy_compress((char *)src, src_len, snapbuf, &snaplen);
 
-	if (snret == SNAPPY_OK) {
-		if (snaplen + sizeof(size_t) < src_len) {
-			*result_lenp = snaplen + sizeof(size_t);
-			*compression_failed = 0;
-
-			/*
-			 * On decompression, snappy requires an exact compressed
-			 * byte count (the current value of snaplen). WiredTiger
-			 * does not preserve that value, so save snaplen at the
-			 * beginning of the destination buffer.
-			 *
-			 * Store the value in little-endian format.
-			 */
+	if (snret == SNAPPY_OK && snaplen + sizeof(size_t) < src_len) {
+		*result_lenp = snaplen + sizeof(size_t);
```

Comment:
A couple of comments here that aren't directly related to your change...

* I think it would be good to have a macro constant `SNAPPY_PREFIX` rather then repeating `sizeof(size_t)` through this code.
* It should also really be a `uint64_t`, since we're assuming that it is 64-bits.